bundles/org.openhab.binding.gpio/src/main/java/org/openhab/binding/gpio/internal/handler/GPIODigitalInputHandler.java

```diff
@@ -0,0 +1,119 @@
+import eu.xeli.jpigpio.PigpioException;
+
+/**
+ * Thing Handler for digital GPIO inputs.
+ *
+ * @author Nils Bauer - Initial contribution
+ */
+@NonNullByDefault
+public class GPIODigitalInputHandler extends GPIOHandler {
+
+    /** The logger. */
+    private final Logger logger = LoggerFactory.getLogger(GPIODigitalInputHandler.class);
+
+    /** The last change. */
+    private Date lastChanged = new Date();
+
+    /**
+     * Instantiates a new GPIO digital input handler.
+     *
+     * @param thing the thing
+     */
+    public GPIODigitalInputHandler(Thing thing) {
+        super(thing, THING_TYPE_DIGITAL_INPUT_CHANNEL, true);
+    }
+
+    @Override
+    public void initialize() {
+        super.initialize();
+
+        GPIOInputConfiguration config = getConfigAs(GPIOInputConfiguration.class);
+        Integer debouncingTime = config.debouncingTime;
+        if (debouncingTime == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "Debouncing time not set.");
+            return;
+        }
+        try {
+            JPigpio jPigpio = this.jPigpio;
+            GPIO gpio = this.gpio;
+            if (jPigpio == null) {
+                logger.warn("Cannot setup gpio change alert because jPigpio is null");
+                return;
+            }
+            if (gpio == null) {
+                logger.warn("Cannot setup gpio change alert because gpio is null");
+                return;
+            }
+            jPigpio.gpioSetAlertFunc(gpio.getPin(), new Alert() {
+
+                @Override
+                public void alert(int gpio, int level, long tick) {
+                    lastChanged = new Date();
+
+                    Date thisChange = new Date();
+                    scheduler.schedule(() -> {
+                        try {
+                            // Check if value changed over time
+                            if (!thisChange.before(lastChanged)) {
+                                Channel channel = getThing().getChannel(THING_TYPE_DIGITAL_INPUT_CHANNEL);
+                                if (channel == null) {
+                                    logger.warn("Cannot find Channel: " + THING_TYPE_DIGITAL_INPUT_CHANNEL);
+                                    return;
+                                }
+                                OnOffType value = getValue();
+                                if (value == null) {
+                                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                                            "GPIO is null");
+                                } else {
+                                    updateState(channel.getUID(), value);
+                                }
+                            }
+                        } catch (PigpioException e) {
+                            logger.warn("Unknown pigpio exception", e);
+                        }
```

Comment:
Please refactor this to it's own method (e.g. `internalInitialize`) and use `this::methodName` here. In general inline-lambdas should be short, only a few lines.

Reply:
Is the new version ok?